clients/src/main/java/org/apache/kafka/common/internals/Topic.java

```diff
@@ -31,7 +31,7 @@
     public static final Set<String> INTERNAL_TOPICS = Collections.unmodifiableSet(
             Utils.mkSet(GROUP_METADATA_TOPIC_NAME, TRANSACTION_STATE_TOPIC_NAME));
 
-    private static final int MAX_NAME_LENGTH = 249;
+    public static final int MAX_NAME_LENGTH = 249;
```

Comment:
nit: so that we don't have this KIP leaking into other components can we also duplicate the `MAX_NAME_LENGTH` here and `containsValidPattern` to `Named` as well

Reply:
@bbejeck this should OK now. I've update the PR with your feedbacks.